python_modules/dagster/dagster/_core/definitions/partition_mapping.py

```diff
@@ -190,10 +191,19 @@ def get_upstream_mapped_partitions_result_for_partitions(
         current_time: Optional[datetime] = None,
         dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
     ) -> UpstreamPartitionsResult:
-        upstream_subset = upstream_partitions_def.subset_with_all_partitions(
-            current_time=current_time, dynamic_partitions_store=dynamic_partitions_store
+        if dynamic_partitions_store is not None and current_time is not None:
```

Comment:
Can you help me understand why we need this condition?

Reply:
Those parameters are required for the `AllPartitionsSubset` class, but are technically optional for this method (I don't think we ever call it in "real code" without those parameters but it is what it is). So in the case that we don't have those parameters we can just fall back to the old implementation.